### PR TITLE
Support output data to csv

### PR DESCRIPTION
**Pull request by jievince**
:set csv filename
:unset csv

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -45,13 +48,32 @@ func bye(username string, interactive bool) {
 	fmt.Println()
 }
 
-// return , does exit
-func clientCmd(query string) bool {
-	plain := strings.ToLower(strings.TrimSpace(query))
-	if plain == "exit" || plain == "quit" {
-		return true
+// client side cmd, will not be sent to server
+func clientCmd(cmd string) (isLocal, exit bool, err error) {
```
Comment (yixinglu) [inline main.go:52]: where is the `err` used ?

File: printer/datasetprinter.go [hunk 2]
```diff
@@ -9,17 +9,43 @@ package printer
...
 )
 
-func valueToString(value *nebula.Value, depth uint) string {
-	// TODO(shylock) get golang runtime limit
-	if depth == 0 { // Avoid too deep recursive
-		return "..."
+type OutCsv struct {
+	fd       *os.File
+	filename string
+	out      bool
```
Comment (yixinglu) [inline printer/datasetprinter.go:21]: this out variable is not necessary, you can check whether fd is nil.

File: main.go (hunk 1 above)
Comment (jievince) [reply main.go:52]: sorry, it's unused, it should be removed.

File: printer/datasetprinter.go [hunk 3]
```diff
@@ -152,23 +186,31 @@ func valueToString(value *nebula.Value, depth uint) string {
 	return ""
 }
 
-func PrintDataSet(dataset *nebula.DataSet) {
-	writer := table.NewWriter()
-	configTableWriter(&writer)
-
+func (p *DataSetPrinter) PrintDataSet(dataset *nebula.DataSet) {
```
Comment (yixinglu) [inline printer/datasetprinter.go:189]: you can check whether dataset is empty here rather than line 205~206.

File: printer/datasetprinter.go [hunk 4]
```diff
@@ -110,7 +144,7 @@ func valueToString(value *nebula.Value, depth uint) string {
 		var buffer bytes.Buffer
 		buffer.WriteString("[")
 		for _, v := range l.GetValues() {
-			buffer.WriteString(valueToString(v, depth-1))
+			buffer.WriteString(valueToString(v))
```
Comment (Shylock-Hg) [inline printer/datasetprinter.go:147]: Why remove the depth limit?

Comment (yixinglu) [reply printer/datasetprinter.go:147]: We shouldn't print something like `...` to user's csv file.

Comment (Shylock-Hg) [reply printer/datasetprinter.go:147]: You can set the depth to infinity when output to csv.

Comment (jievince) [reply printer/datasetprinter.go:147]: I think printing `...` to console is also unuseful.
When the returned dataset columns are too much to show on the terminal, a messy and crowded terminal output will push the user to output the results to CSV

Comment (Shylock-Hg) [reply printer/datasetprinter.go:147]: It's to avoid stack overflow.

Comment (yixinglu) [reply printer/datasetprinter.go:147]: It's not the proper solution by limiting the depth of stack, I'll rewrite this value conversion logic to solve the issue.

**Review by yixinglu — APPROVED** [review]

**Pull request merged**